Add web-dev container and update to React 18

Henriquelay opened the pull request:
Resolves #8.

Big size update. :truck: 

This creates a web-development container, with update React 18 and all dependencies, updated and balated `.eslintrc.yaml`.
Lots of things had to change to update to React 18. Mostly, function components now don't return `React.FC`, and instead it all uses a generic `JSX.Element`. Also, no need to import from `React` everytime :tada:.

I had to do both actions at the same time because of some errors I was getting with installing React on the container, and updating solved it (and caused a bunch of new others, obviously).

Also added configs for Thunder (HTTP) Client.

I have continued on the same branch as before because the issue has been reopened, but thinking about it now, maybe it wasn't a good idea. But it's too late.

Henriquelay committed:
chore(server): Add Thunder Client ignore

Henriquelay committed:
docker(web): Web development server

Henriquelay committed:
deps(web): Update packages, fix eslint

Henriquelay committed:
docker(web): Web-dev container

Henriquelay committed:
feat(web): Docker scripts

Henriquelay committed:
feat(server,web): Ignore files that exist in bind only

Henriquelay committed:
style(web): `ESLint --fix`

Henriquelay committed:
deps(web): Finish updating every single dependency

Henriquelay committed:
deps(web): Upgrade to React 18 (again?)

Henriquelay committed:
fix(web): Eslintada braba, éh isto. 🤙

Also add uuid back as dep

Henriquelay committed:
Agora tá quase juro ainda n olha

Henriquelay committed:
Tá atualizado amigo toma

Henriquelay commented:
Shit, I marked to merge to master.

Henriquelay commented:
There is no conflict, GitHub is drunk. 
![image](https://user-images.githubusercontent.com/37563861/171326072-a742445c-4caf-48b9-b33f-03b0a1110cd2.png)

![image](https://user-images.githubusercontent.com/37563861/171326052-2b18b5cb-c0fd-40c1-b65f-db49a7d114c9.png)

AtilioA commented on web/src/routes/index.tsx:
```diff
@@ -1,26 +1,35 @@
-  </Switch>
-)
+  // If autenticated context is present
+  return user ? <Outlet /> : <Navigate to="/login" />;
```
The `/notificacoes` route should also be private. The rest of the pages should only have visual/functional differences, except for Sign(In|Up) which should redirect the user to the dashboard when they're authenticated.

Henriquelay committed:
fix(server): Notification route is private

Henriquelay committed:
fix(docker): Fix command and change restart policy

Henriquelay committed:
feat(server): Using loadRequiredEnv on ormconfig

Henriquelay committed:
deps(server,web): Add `ts-node` and update `typescript`and `td-node-dev`

Also update web's `yarn.lock`

Henriquelay committed:
style(web): Format routes

Sorry, pre-commit for eslint with multiple projects is kinda shit so it's disabled

Henriquelay committed:
Merge remote-tracking branch 'origin/develop' into 8-use-docker-compose-to-run-locally

Henriquelay committed:
fix(project): Actually, I won't ignore thunder client config

Henriquelay committed:
fix(web): Nested routes didn't work zap

Henriquelay committed:
feat(web): Redirect being done on fronend-side

React 18 style!

AtilioA commented:
I don't know if this private stuff is working (couldn't reproduce) but LGTM :+1:

Henriquelay commented:
I'll resolve the merge conflicts later today then we can finally merge it!

Henriquelay committed:
Merge branch 'develop' into 8-use-docker-compose-to-run-locally

Henriquelay commented:
I have merged manually through the GitHub web editor and have not tested yet, so there is _exactly_ zero chance this goes wrong.

AtilioA committed:
fix(server): remove duplicate import

AtilioA committed:
fix(web): re-add new routes

AtilioA committed:
fix(web): update two new pages that were using old React

AtilioA committed:
fix(web): don't use `string | undefined` in request strings

AtilioA committed:
fix(web): make `Link`s absolute

AtilioA commented:
Seems good to me now, but we're missing Toast messages and feedback is totally bonkers right now

Henriquelay commented:
> we're missing Toast messages

Toast messages were never there, were they?

AtilioA commented:
bruh i can't scroll

Henriquelay commented:
I can't take it anymore 🤠
